### DISCUSSION TITLE
File compression API integrated in appwrite storage API

### DISCUSSION THREAD

**Discussion by mbos2** [appwrite/appwrite#2545] [Ideas]
Would it be cool to integrate file compression in appwrite storage API?
All parameters would be optional to use.

- Compress file size on upload 
- In the case of images, compress size, quality, and aspect ratio

Cloud functions can work for this, but I think it would be a time saver if such a thing would exist directly in appwrites storage API.

**Comment by eldadfux** [comment]
Not sure how would this be different from what we already have supported. Today every file being uploaded is being compressed with GZIP compression. The preview endpoint allows you to change the quality of an images, size, and compression (webp is supported). Appwrite 0.13 will introduce extra flexibility by allowing you to set your bucket settings, and choose if you want to compress, or encrypt files in the bucket. Does this help? Would love more info about how you see this feature in case I'm missing something.

**Reply by tannermeade** [reply]
I am a little confused here. I've written several cloud functions that gather data from the database and save a json file of them in storage. When I save the raw json file vs save a gzip compressed file, the displayed file size is about 0.5x when compressed. So whether the file is being naturally compressed or not -- the raw value of compressing a file is important. I believe that gzip twice should result in a 50% compression rate, usually it's only like 10%.

**Reply by eldadfux** [reply]
We do manage auto-compression for files. From 0.13 we'll also support uploading very big files and compression will be limited to files of up to a specific size. The compression limit should be configurable via env var.
